test/k8sT/Policies.go

```diff
@@ -819,3 +909,156 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
+			Containers:    containers,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	svcName := fmt.Sprintf("svc-%s", podName)
+	By(fmt.Sprintf("Creating a service %s for pod %s in namespace %s", svcName, podName, namespace))
+	svc, err := k.CoreV1().Services(namespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: svcName,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: servicePorts,
+			Selector: map[string]string{
+				"pod-name": podName,
+			},
+		},
+	})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	return pod, svc
+}
+
+func cleanupServerPodAndService(k *helpers.Kubectl, pod *v1.Pod, service *v1.Service) {
+	By(fmt.Sprintf("Cleaning up the server %s/%s", pod.Namespace, pod.Name))
+	err := k.CoreV1().Pods(pod.Namespace).Delete(pod.Name, nil)
+	ExpectWithOffset(1, err).To(BeNil(), "Terminating containers are not deleted after timeout")
+
+	By(fmt.Sprintf("Cleaning up the server's service %s/%s", service.Namespace, service.Name))
+	err = k.CoreV1().Services(service.Namespace).Delete(service.Name, nil)
+	ExpectWithOffset(1, err).To(BeNil(), "Terminating containers are not deleted after timeout")
+}
+
+func createNetworkClientPod(k *helpers.Kubectl, ns, podName string, targetService *v1.Service, dPort int) *v1.Pod {
+	pod, err := k.CoreV1().Pods(ns).Create(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			Labels: map[string]string{
+				"pod-name": podName,
+			},
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				{
+					Name:  fmt.Sprintf("%s-container", podName),
+					Image: "busybox",
+					Args: []string{
+						"/bin/sh",
+						"-c",
+						fmt.Sprintf("for i in $(seq 1 5); do wget -T 8 %s.%s:%d -O - && exit 0 || sleep 1; done; exit 1",
+							targetService.Name, targetService.Namespace, dPort),
+					},
+				},
+			},
+		},
+	})
+
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	return pod
+}
+
+// testCanConnect creates and tests if a given pod can, or can not connect,
+// depending on the canConnect value, to a given service on a specific
+// destination port.
+func testCanConnect(k *helpers.Kubectl, ns, podName string, service *v1.Service, dPort int, canConnect bool) {
+	pod := createNetworkClientPod(k, ns, podName, service, dPort)
+	defer func() {
+		By(fmt.Sprintf("Cleaning up the pod %s", podName))
+		err := k.CoreV1().Pods(ns).Delete(pod.Name, nil)
+		ExpectWithOffset(2, err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have been deleted", pod.Name))
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	err := k.WaitForPodExit(ctx, ns, podName)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), fmt.Sprintf("Pod %s should have finished successfully", pod.Name))
+
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+	success, err := k.WaitForPodSuccess(ctx, ns, podName)
+	if canConnect {
```

Comment:
Instead of canConnect could you change this to the assert like here: https://github.com/cilium/cilium/blob/master/test/runtime/ct.go#L165

Reply:
Does it work even if the messages are different? `"Able to connect to service` vs `"Unable to connect to service`